lua/iswap/internal.lua

```diff
@@ -102,36 +112,30 @@ function M.swap_nodes_and_return_new_ranges(a, b, bufnr, should_move_cursor)
     HAS_SWAPPED = true
   end
 
-  local range1 = ts_utils.node_to_lsp_range(a)
```

Comment:
I stole a big part of the body of this function from nvim-treesitter source code, I'm not sure what the modifications here do but could you explain quickly?

Reply:
So, to accommodate multiple swaps in one call (I believe a swap would make the nodes invalid) the function) the function now primarily operates on ranges rather than nodes, with the helper wrapper above. 

Then I noticed the use of `node_to_lsp_range` was a bit redundant? Like it's just giving new names to the same 4 values. I checked to see if it was some off by one adjustment but it's not so I just removed it (which iirc I had to anyway since we don't have a node anymore) and refactored the rest of it to work. 

It looks like a bigger change than it is, it's really just some find-replace jobs and changing `:range()` to unpack

Reply:
makes sense okay